packages/core/src/baseCogniteClient.ts

```diff
@@ -126,12 +128,17 @@ export default class BaseCogniteClient {
     this.metadata = new MetadataMap();
     this.loginApi = new LoginAPI(this.httpClient, this.metadataMap);
     this.logoutApi = new LogoutApi(this.httpClient, this.metadataMap);
+    if (options.debug) {
+      logger.attach(this.projectName).enable(this.projectName);
```

Comment:
Why do we want this per `projectName` and not let's say per sdk client instance?

Reply:
right, but we need to identify sdk instance somehow to have a reference inside singleton. If we have logger instance per sdk, as it was before, we need to pass logger to the methods as a separate parameter, which is no so convenient than have singleton instance instead.